### PR TITLE
MULEbot code cleanup & fixes

### LINKED ISSUE #38594: Getting stunned while riding a mulebot doesn't make you fall off
Round ID: 89703 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Mulebots don't respect the normal rules for the rider being stunned and falling off, like all other vehicles.

### LINKED ISSUE #42918: Blood trails look wrong on the horizontal.
Note the horizontal trail acting like it's trying to go up and down as well.

### LINKED ISSUE #46030: Mulebots don't change the pixel_y after the first time.
## Round ID:

local updated master

## Testmerges:
N/A

## Reproduction:

Get on a mulebot
Get off the mulebot
Get on the mulebot again, you will now not be raised up on the mulebot.
video: [here](https://youtu.be/F84tmdwir18)

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
refactor: Optimized a good chunk of MULEbot code, fixing things such as it rebuilding it's overlays/icons 5 times every second while moving to a destination. Less free lag!
fix: Fixed MULEbots closing crates when the crate doesn't actually get loaded onto it.
fix: Fixed the maintenance hatch on MULEbots visually closing for a moment when emagged.
fix: Paranormal MULEbots will no longer briefly turn into a normal MULEbot when emagged.
fix: Fixed interface exploits involving paranormal MULEbots carrying ghosts.
fix: Fixed MULEbot blood tracks sometimes facing the wrong direction.
fix: Player controlled MULEbots will now leave bloody tracks if they've run over a mob.
fix: Fixed players riding MULEbots sometimes appearing underneath it instead of ontop of it.
fix: Fixed AI's making a chiming noise when they receive a remote notification from a MULEbot. Now only the AI will hear the notification's sound.
fix: Fixed MULEbots dropping mobs off at their current location instead of at their dropoff point.
fix: Fixed a minor runtime related to MULEbots dropping off deleted items/mobs.
fix: Fixed mobs having the wrong height offset when riding MULEbots.
fix: Fixed mobs sometimes ending up underneath MULEbots while riding them.
fix: Fixed mobs not falling off MULEbots when stunned / incapacitated. 
fix: Fixed inducers not recharging MULEbots.
fix: Fixed MULEbots leaving bloody tracks in space.
fix: Fixed MULEbots leaving multiple blood tracks in the same spot if they failed to move.
fix: Fixed hacked MULEbots spamming messages about knocking down a mob when they didn't actually do it.
fix: Fixed MULEbot commands not resetting when they were turned off.
fix: Fixed MULEbots not turning off properly when their maintenance hatch is opened.
fix: Fixed MULEbots not turning off properly when they ran out of power while moving.
fix: Fixed MULEbots flashing their external warning lights while they're turned off when their avoidance wire is cut.
fix: Fixed mobs being able to walk right over MULEbots when they turn off.
fix: MULEbots will now try to put their powercell in your hands when you remove them with a crowbar.
fix: Fixed the interface for MULEbots briefly looking like you turned it on successfully even though it was out of power.
fix: MULEbots will now provide feedback if they fail to turn on due to their maintenance panel being open.
fix: MULEbots will no longer flash and beep when their wires are pulsed if they don't have power.
fix: MULEbots will now actually flash when their external warning lights are pulsed.
tweak: MULEbots will now flash when they beep / buzz.
add: Added a bit more examine text to MULEbots.
/:cl:

Fixes #42918
Fixes #46030
fixes #38594

### CODE REVIEW

**Comment by optimumtact** [comment]
Okay, i'm impressed

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 1]
```diff
@@ -779,74 +782,59 @@
...
 	update_icon()
-	if(istype(AM, /mob/dead/observer))
-		ghost_rider = TRUE
-		RegisterSignal(AM, COMSIG_MOVABLE_MOVED, .proc/ghostmoved)
 
-/mob/living/simple_animal/bot/mulebot/paranormal/update_icon()
-	if(load && isobserver(load) && isnull(ghost_overlay))//there are issues with adding a ghost as an overlay, and this prevents metagaming to see who is dead
-		ghost_rider = TRUE
-		visible_message("<span class='warning'>A ghostly figure appears on [src]!</span>")
-		ghost_overlay = ghost_overlay || mutable_appearance('icons/mob/mob.dmi')
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:821]:
```
if(isnull(ghost_overlay))
	ghost_overlay = ghost_overlay
```

lol

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 2]
```diff
@@ -70,17 +70,49 @@
 	mulebot_count += 1
 	set_id(suffix || id || "#[mulebot_count]")
 	suffix = null
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:73]: this was loaded in simple/bot's buckle_mob, so no change in adding it in Initialize aside from the relevant fixes.

**Comment by ShizCalev** [comment]
Tested quite thoroughly. Squashed the last bug I can find with them, so should be good to go!

**Comment by ShizCalev** [comment]
Tested thoroughly, all good to go.

Identified a number of other bugs through this with the riding component / buckling handling, opened PR's for those.

**Pull request merged**